console_game.rb

```diff
@@ -1,40 +1,45 @@
-def main()
-    puts "Game: HiLo"
-
-    #computer chooses a number from 1 to 100 (inclusive)
-    comp_choice = rand(101)
-    
-    number_of_choices = 1
-    
-   # do-while loop for each of the 6 chances 
-   loop do
-        #gets users guess
-        puts " \n Enter your guess number " + number_of_choices.to_s
-        user_choice  = gets.chomp
-        
-        #checks if not nil
-        user_choice != "" && user_choice.match?(/\A-?\d+\Z/) ? user_choice = user_choice.to_i : (puts "Please enter a valid integer value" ; next)
-
-        if user_choice == comp_choice 
-            puts "you guessed the number correctly"
-            break
-        
-        elsif user_choice < comp_choice
-            puts "you guessed less that the required number"
-        
-        else
-            puts "you guessed more than the required number"
-        end
-
-        if number_of_choices > 5
-            puts "You've run out of chances to guess. Better luck next time"
-            puts "The number was: " + comp_choice.to_s
-            break
-        end
-        
-        number_of_choices += 1
+# rubocop:disable Metrics/AbcSize, Style/FrozenStringLiteralComment, Metrics/MethodLength
+def main
+  puts 'Game: HiLo'
+
+  # computer chooses a number from 1 to 100 (inclusive)
```

Comment:
```suggestion
  # computer chooses a number from 0 to 100 (inclusive)
```
I think `rand` could be 0.

Reply:
Thanks. Have used the range specified to ensure `rand` chooses a valid number.